projects/ngx-d3timeline/src/lib/resources-axis/resources-axis.service.ts

```diff
@@ -0,0 +1,84 @@
+import { DataService } from '../data.service';
+import { OptionsService } from '../options.service';
+import { ViewService } from '../view/view.service';
+import { combineLatest } from 'rxjs';
+import { AxisViewModel, TickInfo, Line } from '../axis-view-model';
+import { Orientation } from '../orientation';
+import { TimelineView } from '../view/timeline-view';
+import { map } from 'rxjs/operators';
+import { ScaleService } from '../scale.service';
+import { ScaleBand } from 'd3-scale';
+import { Injectable } from '@angular/core';
+import { TimelineEvent } from '../timeline-event';
+
+@Injectable({ providedIn: 'root' })
+export class ResourcesAxisService {
+  vm$ = combineLatest([
+    this.dataService.data$,
+    this.optionsService.orientation$,
+    this.viewService.view$
+  ]).pipe(
+    map(([data, orientation, view]) =>
+      this.createAxisViewModel(data, orientation, view)
+    )
+  );
+
+  constructor(
+    private dataService: DataService,
+    private optionsService: OptionsService,
+    private viewService: ViewService,
+    private scaleService: ScaleService
+  ) {}
+
+  private createAxisViewModel(
+    data: TimelineEvent[],
+    timelineOrientation: Orientation,
+    view: TimelineView
+  ): AxisViewModel {
+    const orientation = this.optionsService.flipOrientation(
+      timelineOrientation
+    );
+
+    const scale = this.scaleService.configureScaleBand(data, view, orientation);
+
+    return {
+      tickInfos: this.getTickInfos(scale, orientation),
+      axisLine: this.getAxisLine(scale, orientation)
+    };
+  }
+
+  private getTickInfos(
+    scale: ScaleBand<string>,
+    orientation: Orientation
+  ): TickInfo[] {
+    return scale.domain().map(tick => ({
```

Comment:
domain() is not returning an array of **tick**s
**value** while generic is more suitable. Or maybe **category** but maybe that is too specific.

```suggestion
    return scale.domain().map(value => ({
```

Reply:
I think leave this one for another case, there are a quite a few places it would need changing

Reply:
I've made the change in this file. Over the next few issues I'll be touching all the other places where it needs to be changed, so will keep an eye out for it.